src/Controls/src/Core/BindableObject.cs

```diff
@@ -283,7 +283,7 @@ internal void SetBinding(BindableProperty targetProperty, BindingBase binding, S
 			{
 				var kvp = context.Values.Last();
 				context.Values.Remove(kvp.Key);
-				context.Values.Add(SetterSpecificity.FromBinding, kvp.Value);
+				context.Values[SetterSpecificity.FromBinding] = kvp.Value;
```

Comment:
To test it can use any Style with a Binding property?

Reply:
In this case, without a test case, we do not know if the issue is related setter and style. that hypothesis was only me guessing. So I can't add a test for this issue as we don't know what the issue is.

What we have is a stack trace, and we can (and did) fix the code so that error no longer fire